### DISCUSSION TITLE
{Alert} An "ifLet" at "/ContentView.swift:132" received a presentation action when destination state was absent.

### DISCUSSION THREAD

**Discussion by anansadiya94** [pointfreeco/swift-composable-architecture#2518] [General]
I'm using the new navigation patterns for alerts and receiving a run time warning after tapping the exit button in the alert.
I thought it might have to do with my project, so I created a new project, and here is the code:

```
import SwiftUI
import ComposableArchitecture

struct Feature: Reducer {
    struct State: Equatable {
        @PresentationState var alert: AlertState<AlertAction>?
    }
    
    enum Action: Equatable {
        case exitTapped
        case alert(PresentationAction<AlertAction>)
    }
    
    enum AlertAction: Equatable {
        case exitConfirmed
    }
    
    var body: some Reducer<State, Action> {
        Reduce { state, action in
            switch action {
            case .exitTapped:
                state.alert = .init(
                    title: {
                        TextState("Are you sure you want to exit?")
                    },
                    actions: {
                        ButtonState(role: .destructive, action: .exitConfirmed) {
                            TextState("Yes")
                        }
                    }
                )
                return .none
            case .alert(.presented(.exitConfirmed)):
                print("Exit confirmed!")
                return .none
                
            case .alert:
                return .none
            }
        }
        .ifLet(\.$alert, action: /Action.alert)
    }
}
    

struct ContentView: View {
    let store: StoreOf<Feature> = Store(initialState: Feature.State(), reducer: { Feature()._printChanges() })
    var body: some View {
        WithViewStore(store, observe: { $0 }) { viewStore in
            VStack {
                Button {
                    viewStore.send(.exitTapped)
                } label: {
                    Text("Exit!")
                }
            }
        }
        .alert(store: store.scope(state: \.$alert, action: { .alert($0) }))
    }
}

#Preview {
    ContentView()
}
```
I'm using the 1.2.0 TCA version and it's reproducible with both Xcode 14.3.1 and 15.0.
Please let me know what I'm doing wrong.
Thanks,

**Comment by mbrandonw** [comment]
Hi @anansadiya94, I am not able to reproduce this problem. I copied and pasted your code into a project and it seems to work fine with no runtime warning. Could you package up a project that reproduces the problem?

**Reply by anansadiya94** [reply]
Hi @mbrandonw , I tested it again and it seems to be reproducible on iOS 15 only. Sorry for not mentioning it before, but I was using my iOS 15 testing device and caught the runtime warning there. Please give it a try running it on iOS 15 simulator, I used `iPhone SE (3rd generation) 15.5`. And keep me posted.
Attaching the [Project](https://github.com/pointfreeco/swift-composable-architecture/files/12907247/Challenge1.zip), if you still need it.
Thanks,

**Reply by stephencelis** [reply]
@anansadiya94 iOS 15 alerts appear to write dismissal to their bindings _before_ executing their action closure, which is why you see this discrepancy. I'm not sure it's worth working around in our library, though, since it technically is revealing a potential issue, like if the action depended on the current state of the alert, which would be `nil` at that moment. It's all a bit tricky, but at least the behavior has improved in iOS 16. I think you can safely ignore the warning we emit in debug iOS 15 builds, though! It's just good to be aware of what the warning means.
